### ISSUE TITLE
[Bug] Weird Hair Appearing in Path

### ISSUE THREAD

**Issue by pyj-pd** [steveruizok/perfect-freehand#85]
Hi! I recently found this project, and I am loving it so much.

However, I found a small bug where the path shows weird “hair” when strokes get intersected to each other.

Please refer to the video and image below. Thank you.

https://github.com/user-attachments/assets/811d4241-7bf9-47c9-8363-3a394fa539e2

![Capture of the video above](https://github.com/user-attachments/assets/cd8b5a88-c052-4ffc-8655-e071e4ea0087)

![When stroke size is increased, it gets worse](https://github.com/user-attachments/assets/a7d8a799-4521-4ddc-8d45-83445a41908e)

**Comment by steveruizok** [comment]
Yep yep, the density of "turns" can lead to these bugs, ie when the line goes south south, then south north south north south north. It's a problem we've actually solved on tldraw.com's freehand library by separating the line into separate lines when sharp corners occur.

**Comment by muzakon** [comment]
Hello, any update? This problem still occurs.

**Comment by steveruizok** [comment]
Thanks for the detailed report! The "hair" you're seeing is expected behavior when a stroke rapidly changes direction.

Closing as this is working as designed - the self-intersections are inherent to how the polygon outline is generated. FWIW in tldraw we addressed this on the rendering side by rendering different strokes at sharp corners, rather than trying to handle them in the polygons.
